contracts/DaiHrd.sol

```diff
@@ -486,38 +479,38 @@ contract DaiHrd is ERC777, MakerFunctions {
 		}
 	}
 
-	function deposit(uint256 attodai) external returns(uint256 depositedAttopot) {
+	function deposit(uint256 attodai) external returns(uint256 attodaiHrd) {
 		dai.transferFrom(msg.sender, address(this), attodai);
 		daiJoin.join(address(this), dai.balanceOf(address(this)));
-		depositedAttopot = depositVatDaiForAccount(msg.sender);
+		uint256 depositedAttopot = depositVatDaiForAccount(msg.sender);
 		emit Deposit(msg.sender, attodai, depositedAttopot);
 		return depositedAttopot;
 	}
 
-	// If the user has vat dai directly (after performing vault actions, for instance), they don't need to create the DAI ERC20 for us to burn it, we'll accept vat dai
-	function depositVatDai(uint256 attoVatDai) external returns(uint256 depositedAttopot) {
-		uint256 attorontodai = attoVatDai.mul(10 ** 27);
-		vat.move(msg.sender, address(this), attorontodai);
-		depositedAttopot = depositVatDaiForAccount(msg.sender);
-		emit DepositVatDai(msg.sender, attorontodai, depositedAttopot);
+	// If the user has vat dai directly (after performing vault actions, for instance), they don't need to create the DAI ERC20 just so we can burn it, we'll accept vat dai
+	function depositVatDai(uint256 attorontovatDai) external returns(uint256 attodaiHrd) {
+		vat.move(msg.sender, address(this), attorontovatDai);
+		uint256 depositedAttopot = depositVatDaiForAccount(msg.sender);
+		emit DepositVatDai(msg.sender, attorontovatDai, depositedAttopot);
 		return depositedAttopot;
 	}
 
 	function withdrawTo(address recipient, uint256 attodaiHrd) external returns(uint256 attodai) {
-		pot.drip();
-		return _withdraw(recipient, attodaiHrd);
+		// Don't need rontodaiPerPot, so we don't call updateAndFetchChi
+		if (pot.rho() != now) pot.drip();
```

Comment:
This was missing the optimization that other similar functions had, so copied it here for consistency.

Reply:
Sounds good. This is the only case where we want to drip, but don't actually need the value of chi